src/spacy/app.clj

```diff
@@ -84,12 +93,28 @@
                    (let [current-user "joy"
                          slug (get-in ctx [:parameters :path :event-slug])
                          params  (get-in ctx [:parameters :form])
-                         session (data/fetch data slug)
-                         new-state (domain/suggest-session session current-user params)
-                         channel (:channel events)]
+                         state (data/fetch data slug)
+                         new-state (domain/suggest-session state current-user params)]
                      (data/persist! data new-state)
                      (yada-redirect ctx (bidi/path-for routes ::event :event-slug slug))))}}})))
 
+(defn schedule-session [{:keys [data events]}]
+  (yada/handler
+   (yada/resource
+    {:methods
+     {:post
+      {:consumes "application/x-www-form-urlencoded"
+       :parameters {:form {:room String :time String :id java.util.UUID}}
+       :response (fn [ctx]
+                   (let [slug (get-in ctx [:parameters :path :event-slug])
+                         params (get-in ctx [:parameters :form])
+                         state (data/fetch data slug)]
+                     (if-not (domain/can-schedule-session? state params)
+                       (reject-request ctx)
+                       (let [new-state (domain/schedule-session state params)]
+                         (data/persist! data new-state)
+                         (yada-redirect ctx (bidi/path-for routes ::event :event-slug slug))))))}}})))
+
```

Comment:
The `if-not` logic makes me wonder if we need to think about handling edge cases and errors. Maybe domain functions returning `::domain/error` with a keyword identifying the problem? Then the UI could decide how to present it, maybe even with i18n.

Reply:
Hmm... that it an interesting idea.